### PR TITLE
TIP-860: Move repositories/cursor/es stuuf from enrich bundle to the right BC

### PR DESCRIPTION
**Pull request by arnolanglade**
**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | yes
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: CHANGELOG-3.0.md [hunk 1]
```diff
@@ -893,3 +893,25 @@
 - Move `Pim\Bundle\EnrichBundle\Normalizer\VersionedAttributeNormalizer` to `Akeneo\Pim\Structure\Component\Normalizer\InternalApi\VersionedAttributeNormalizer`
 - Move `Pim\Bundle\EnrichBundle\Normalizer\FileNormalizer` to `Akeneo\Tool\Component\FileStorage\Normalizer\FileNormalizer`
 - Move `Pim\Bundle\EnrichBundle\Normalizer\DatagridViewNormalizer` to `Oro\Bundle\PimDataGridBundle\Normalizer\InternalApi\DatagridViewNormalizer`
+- Move `Pim\Bundle\EnrichBundle\Doctrine\ORM\Repository\ChannelRepository` to `Akeneo\Channel\Component\Repository\InternalApi\ChannelRepository`
+- Move `Pim\Bundle\EnrichBundle\Doctrine\ORM\Repository\CurrencyRepository` to `Akeneo\Channel\Component\Repository\InternalApi\CurrencyRepository`
+- Move `Pim\Bundle\EnrichBundle\Doctrine\ORM\Repository\LocaleRepository` to `Akeneo\Channel\Component\Repository\InternalApi\LocaleRepository`
+- Move `Pim\Bundle\EnrichBundle\Cursor\SequentialEditProduct` to `Akeneo\Pim\Enrichment\Bundle\Cursor\SequentialEditProduct`
+- Move `Pim\Bundle\EnrichBundle\Cursor\SequentialEditProduct` to `Akeneo\Pim\Enrichment\Bundle\Cursor\SequentialEditProduct`
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply CHANGELOG-3.0.md]: Fixed https://github.com/akeneo/pim-community-dev/pull/8864/files#diff-8ee894204f471eae7d606c828a1af52fL900

File: src/Akeneo/Channel/Bundle/Doctrine/Repository/InternalApi/LocaleRepository.php [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace Pim\Bundle\EnrichBundle\Doctrine\ORM\Repository;
+namespace Akeneo\Channel\Component\Repository\InternalApi;
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Akeneo/Channel/Bundle/Doctrine/Repository/InternalApi/LocaleRepository.php]: https://github.com/akeneo/pim-community-dev/pull/8864/files#diff-e2a10d7e213b3677c10c6a52fbafe424R3

File: src/Akeneo/Channel/Bundle/Doctrine/Repository/InternalApi/CurrencyRepository.php [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace Pim\Bundle\EnrichBundle\Doctrine\ORM\Repository;
+namespace Akeneo\Channel\Component\Repository\InternalApi;
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Akeneo/Channel/Bundle/Doctrine/Repository/InternalApi/CurrencyRepository.php]: https://github.com/akeneo/pim-community-dev/pull/8864/files#diff-bb1fa22464642458e344d8579f096dbaR3

File: src/Akeneo/Channel/Bundle/Doctrine/Repository/InternalApi/ChannelRepository.php [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace Pim\Bundle\EnrichBundle\Doctrine\ORM\Repository;
+namespace Akeneo\Channel\Component\Repository\InternalApi;
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Akeneo/Channel/Bundle/Doctrine/Repository/InternalApi/ChannelRepository.php]: https://github.com/akeneo/pim-community-dev/pull/8864/files#diff-37a8986efd1bdb37e7b1388fe6133205R3

File: CHANGELOG-3.0.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (arnolanglade) [reply CHANGELOG-3.0.md]: Fixed https://github.com/akeneo/pim-community-dev/pull/8864/files#diff-8ee894204f471eae7d606c828a1af52fL900https://github.com/akeneo/pim-community-dev/pull/8864/files#diff-8ee894204f471eae7d606c828a1af52fL900

File: src/Akeneo/Pim/Enrichment/Bundle/spec/Elasticsearch/Sorter/InGroupSorterSpec.php [hunk 5]
```diff
@@ -89,7 +89,7 @@ function it_throws_an_exception_when_group_is_null()
     {
         $this->shouldThrow(
             new InvalidArgumentException(
-                InGroupSorter::class,
+                \Akeneo\Pim\Enrichment\Bundle\Elasticsearch\Sorter\InGroupSorter::class,
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Akeneo/Pim/Enrichment/Bundle/spec/Elasticsearch/Sorter/InGroupSorterSpec.php]: Fixed https://github.com/akeneo/pim-community-dev/pull/8864/files#diff-cd087275a6a96228fe5ade17069f11c2R125

File: src/Akeneo/Pim/Enrichment/Bundle/spec/Elasticsearch/Sorter/InGroupSorterSpec.php [hunk 6]
```diff
@@ -122,7 +122,7 @@ function it_throws_an_exception_when_the_directions_does_not_exist(
         $this->shouldThrow(
             InvalidDirectionException::notSupported(
                 'A_BAD_DIRECTION',
-                InGroupSorter::class
+                \Akeneo\Pim\Enrichment\Bundle\Elasticsearch\Sorter\InGroupSorter::class
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Akeneo/Pim/Enrichment/Bundle/spec/Elasticsearch/Sorter/InGroupSorterSpec.php]: Fixed https://github.com/akeneo/pim-community-dev/pull/8864/files#diff-cd087275a6a96228fe5ade17069f11c2R125

File: src/Akeneo/Pim/Enrichment/Bundle/spec/Doctrine/ORM/Repository/InternalApi/GroupRepositorySpec.php [hunk 7]
```diff
@@ -25,7 +25,7 @@ function let(UserContext $userContext, EntityManager $em, ClassMetadata $classMe
 
     function it_is_initializable()
     {
-        $this->shouldHaveType(GroupRepository::class);
+        $this->shouldHaveType(\Akeneo\Pim\Enrichment\Bundle\Doctrine\ORM\Repository\InternalApi\GroupRepository::class);
```
Comment (willy-ahva) [inline src/Akeneo/Pim/Enrichment/Bundle/spec/Doctrine/ORM/Repository/InternalApi/GroupRepositorySpec.php]: use `use` pls

**Pull request merged**